File: qb-network-graph-ui/src/api/matching.js
import { apiClient } from './client';

export async function getPendingMatches() {
  return apiClient.get('/matching/pending');
}

export async function getCandidates(matchId) {
  return apiClient.get(`/matching/${matchId}/candidates`);
}

export async function resolveMatch(matchId, resolution, candidateGoldenId = null) {
  const body = { resolution };
  if (candidateGoldenId) body.candidateGoldenId = candidateGoldenId;
  return apiClient.post(`/matching/${matchId}/resolve`, body);
}

/**
 * Run entity resolution against a manual input.
 * Returns tier (0=no match, 1=deterministic, 2=AI candidates) + results.
 */
export async function resolveEntity(input) {
  return apiClient.post('/matching/resolve', input);
}

File: qb-network-graph-be/routes/v1/matching.py
"""Matching routes: GET /matching/pending, GET /matching/{id}/candidates, POST /matching/{id}/resolve, POST /matching/resolve."""
from fastapi import APIRouter, Depends, HTTPException

from dependencies import get_matching_service, get_search_service
from models.api import ResolveRequest, AdHocResolveRequest
from services.matching_service import MatchingService
from services.search_service import SearchService

router = APIRouter()


@router.get("/matching/pending")
def get_pending_matches(svc: MatchingService = Depends(get_matching_service)):
    return svc.get_pending()


@router.get("/matching/{match_id}/candidates")
def get_candidates(
    match_id: str,
    svc: MatchingService = Depends(get_matching_service),
):
    try:
        return svc.get_candidates(match_id)
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Candidate lookup failed: {e}")


@router.post("/matching/{match_id}/resolve")
def resolve_match(
    match_id: str,
    body: ResolveRequest,
    svc: MatchingService = Depends(get_matching_service),
):
    try:
        return svc.resolve(match_id, body.resolution, candidate_override_id=body.candidateGoldenId)
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Resolution failed: {e}")


@router.post("/matching/resolve")
def resolve_adhoc(
    body: AdHocResolveRequest,
    svc: SearchService = Depends(get_search_service),
):
    return svc.resolve_adhoc(
        name=body.name, ein=body.ein,
        city=body.city, state=body.state,
        industry=body.industry,
    )
